docs/pages/project/contributing/contributing-error.md

```diff
@@ -9,27 +9,127 @@ category: contributing
 ---
 
 Meshery pervasively uses MeshKit as a golang and service mesh management-specific library in all of its components. MeshKit helps populate error messages with a uniform and useful set of informative attributes. 
+
+This tool analyzes, verifies and updates error codes in Meshery source code trees. It extracts error details into a file that can be used for publishing all error code references on the Meshery website. The objective to create this was to avoid centralized handling of error codes, automating everything
+
+Inorder to create a meshery error object, you will need to create a custom wrapper object for the native golang error. This can be done from the <a href="https://github.com/layer5io/meshkit/tree/master/errors">Meshkit Error</a> package. 
+
+Use the `errors.New()` function to create a new instance of the error object and pass situation-specific attributes as function arguments. 
 The below are the listed attributes:
 - Code
 - Short Description
 - Long Description
 - Probable Cause
 - Suggested Remediation
 
-Inorder to create a meshery error object, you will need to create a custom wrapper object for the native golang error. This can be done from the <a href="https://github.com/layer5io/meshkit/tree/master/errors">Meshkit Error</a> package. 
+### Syntax
+     errors.New(ErrExampleCode, errors.Alert, []string{"<short-description>"}, []string{"<long-description>"}, []string{"<probable-cause>"}, []string{"<suggested remediation>"})
+
 
-Use the `errors.New()` function to create a new instance of the error object and pass situation-specific attributes as function arguments. See the following example for reference.
+It is intended to be run locally and as part of a CI workflow.
+
+- Errors names and codes are namespaced to components, i.e. they need to be unique within a component, which is verified by this tool.
+- A component corresponds usually to a repository. Components have a type and a name. 
+  They are also returned from the ComponentInfo endpoint, e.g. for adapters.
```

Comment:
Is there a ComponentInfo endpoint? :thinking:

Reply:
No I think.

Reply:
@DelusionalOptimist @sayantan1413 yes there is for adapters (https://github.com/meshery/meshery/blob/b04208b6614250ac723317e3040b079c93139800/meshes/meshops.proto#L17) and eventually, there should be for all components. It has a component name and type field and which should be the same  as component name and info in the component_info.json file. My idea was that this was picked up for all functionality from `component_info.json` file, so that it was guaranteed to be consistent. Eventually, component name and type should be part of any message / notification emitted by any component.

Reply:
@mgfeller is there anything you want me to add or remove from [here](https://60f2beb4f706dbe8151b6a6c--meshery-docs-preview.netlify.app/project/contributing-error) : )

Reply:
I think this sentence could be removed: "They are also returned from the ComponentInfo endpoint, e.g. for adapters."

Reply:
Otherwise, not quite sure anything should be removed or added. Personally, I always like to have enough information so that I understand how the dots connect, so I'm biased. But I understand the concern that some can be confusing. What do you think @Shreyas220 ? Anything superfluous here? 

If not, I think we can just go forward and update later...